sonic-chassisd/scripts/chassisd

```diff
@@ -0,0 +1,334 @@
+
+try:
+    from sonic_platform_base.module_base import ModuleBase
+except ImportError as e:
+    from tests.mock_module_base import ModuleBase
+
+#
+# Constants ====================================================================
+#
+
+SYSLOG_IDENTIFIER = "chassisd"
+
+CHASSIS_CFG_TABLE = 'CHASSIS_MODULE'
+
+CHASSIS_INFO_TABLE = 'CHASSIS_TABLE'
+CHASSIS_INFO_KEY_TEMPLATE = 'CHASSIS {}'
+CHASSIS_INFO_CARD_NUM_FIELD = 'module_num'
+
+CHASSIS_MODULE_INFO_TABLE = 'CHASSIS_MODULE_TABLE'
+CHASSIS_MODULE_INFO_KEY_TEMPLATE = 'CHASSIS_MODULE {}'
+CHASSIS_MODULE_INFO_NAME_FIELD = 'name'
+CHASSIS_MODULE_INFO_DESC_FIELD = 'desc'
+CHASSIS_MODULE_INFO_SLOT_FIELD = 'slot'
+CHASSIS_MODULE_INFO_OPERSTATUS_FIELD = 'oper_status'
+
+CHASSIS_INFO_UPDATE_PERIOD_SECS = 10
+
+CHASSIS_LOAD_ERROR = 1
+CHASSIS_NOT_SUPPORTED = 2
+
+platform_chassis = None
+
+SELECT_TIMEOUT = 1000
+
+NOT_AVAILABLE = 'N/A'
+INVALID_SLOT = ModuleBase.MODULE_INVALID_SLOT
+INVALID_MODULE_INDEX = -1
+
+MODULE_ADMIN_DOWN = 0
+MODULE_ADMIN_UP   = 1
+
+#
+# Helper functions =============================================================
+#
+
+# try get information from platform API and return a default value if caught NotImplementedError
+def try_get(callback, *args, **kwargs):
+    """
+    Handy function to invoke the callback and catch NotImplementedError
+    :param callback: Callback to be invoked
+    :param default: Default return value if exception occur
+    :return: Default return value if exception occur else return value of the callback
+    """
+    default = kwargs.get('default', NOT_AVAILABLE)
+    try:
+        ret = callback(*args)
+        if ret is None:
+            ret = default
+    except NotImplementedError:
+        ret = default
+
+    return ret
+
+#
+# Module Config Updater ========================================================
+#
+class ModuleConfigUpdater(logger.Logger):
+
+    def __init__(self, log_identifier, chassis):
+       """
+       Constructor for ModuleConfigUpdater
+       :param chassis: Object representing a platform chassis
+       """
+       super(ModuleConfigUpdater, self).__init__(log_identifier)
+
+       self.chassis = chassis
+
+    def deinit(self):
+        """
+        Destructor of ModuleConfigUpdater
+        :return:
+        """
+
+    def module_config_update(self, key, admin_state):
+        if not key.startswith(ModuleBase.MODULE_TYPE_SUPERVISOR) and \
+           not key.startswith(ModuleBase.MODULE_TYPE_LINE) and \
+           not key.startswith(ModuleBase.MODULE_TYPE_FABRIC):
+            self.log_error("Incorrect module-name {}. Should start with {} or {} or {}".format(key,
+                           ModuleBase.MODULE_TYPE_SUPERVISOR, ModuleBase.MODULE_TYPE_LINE,
+                           ModuleBase.MODULE_TYPE_FABRIC))
+            return
+
+        module_index = try_get(self.chassis.get_module_index, key, default=INVALID_MODULE_INDEX)
+
+        #Continue if the index is invalid
```

Comment:
Please add a space after `#` for all comments through out the PR per PEP8 standards.

Reply:
Fixed.